enhancements/update/duration-of-migration-to-multi-arch.md

```diff
@@ -0,0 +1,166 @@
+
+### Non-Goals
+
+Whether or not 'oc adm upgrade' is executed succesfully or the requested upgrade is accepted is not in the scope of this enhancement. So it is _not_ required to have test cases for this enhancement to cover (although they are valid in general):
+- upgrade from multi-arch to single-arch,
+- upgrade from multi-arch to multi-arch,
+- upgrade a cluster in a cloud provider where multi-arch is not supported to multi-arch, or
+- cover the rescue path in case migration to multi-arch hits a failure.
+
+The `oc adm upgrade status` might be useful to monitor the migration progress to multi-arch but the output of the command does not block this enhancement and its implementation. The `status` command is designed for cluster upgrades but migration is not an upgrade conceptually and is implemented through an upgrade.
+
+## Proposal
+
+The proposal includes the steps that an OpenShift cluster operator may take to provide information about the upgrade in additional to the existing names and version numbers for the components that the operator manages.
+
+* Add a few elements into the existing list `ClusterOperator.status.versions` in the `ClusterOperator` manifest. Those new `versions` contains the image pull specifications for the operator and OPTIONALLY the managed operands that the operator chooses to track. Although the version stays the same, the image pull specification are different if a cluster becomes multi-arch.
+
+* Populate the `ClusterOperator.status.versions` including these new elements above.
+
+The CVO has already implemented the way that it checks if each element in the `ClusterVersion.status.versions` from the payload's has the counterpart in the actual one on the cluster.
+
+There is no need to extend the APIs in `openshift/api`, or modify the CVO implementation.
+
+### Workflow Description
+
+This enhancement changes no workflow.
+
+### API Extensions
+
+None.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+[Hosted control planes enables the decoupling of updates between the control plane and the data plane](https://docs.openshift.com/container-platform/4.17/hosted_control_planes/hcp-updating.html#hcp-get-upgrade-versions_hcp-updating):
+
+* Update a control plane: specify `HostedCluster.spec.release`, and
+* Update a data plane: specify `NodePool.spec.release`,
+
+where the value is [the image pull specification of an OCP release payload image](https://hypershift-docs.netlify.app/reference/api/#hypershift.openshift.io/v1beta1.Release).
+
+The MCO does not run on a hosted cluster and thus our proposal in this enhancement makes no improvement (or regression) for migration of a hosted cluster to multi-arch. Currently there is no way to discover if a hosted cluster is ready to serve a cross-arch node pool. The requests are mainly from the users of standalone clusters at the moment. When hosted cluster users desire the improvement, we can work with the HyperShift team for a solution.
+
```

Comment:
cc @jeffdyoung this might be needed sooner than later as HCP is in the process of using multi payload as the default

Reply:
This is already the case. All new ROSA HCP clusters are multi-arch by default. ARO HCP is planning to do this as well.

Reply:
Do I understand this correct?
- Migration of existing hosted clusters from ROSA to multi-arch has done already. The new ones are multi-arch by default.
- ARO will catch up with ROSA soon.

Considering ROSA has done it without our fix, ARO is expected to be done smoothly as well.

Then the solution for HCP is not that demanding at the moment which gives us some air to do it in a later phase, e.g., after we handle the standalone clusters.